File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-1.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int row = matrix.length;
        int col = matrix[0].length;
        int row_arr[] = new int[row];
        int col_arr[] = new int[col];
        for(int i=0; i<row; i++)
        {
            for(int j=0; j<col; j++)
            {
                if(matrix[i][j]==0)
                {
                    row_arr[i]=1;
                    col_arr[j]=1;
                }
            }
        }
        for(int i=0;i<row;i++)
        {
            for(int j=0; j<col; j++)
            {
                if(row_arr[i]==1)
                {
                    matrix[i][j]=0;
                }
                if(col_arr[j]==1)
                {
                    matrix[i][j]=0;
                }
            }
        }
        
        
    }
}

File: Data Structures & Algorithms/max-area-of-island/submission-1.cpp
class Solution {
public:
    void dfs(int row, int col, vector<vector<int>>&grid, vector<vector<int>>&vis, int &area, int rdel[], int cdel[], int n, int m, int &cnt)
    {
        cnt++;
        vis[row][col]=1;
        for(int i=0;i<4;i++)
        {
            int nr = row+rdel[i];
            int nc = col+cdel[i];
            if(nr<n && nr>=0 && nc<m && nc>=0 && grid[nr][nc]==1 && vis[nr][nc]!=1)
            {
                dfs(nr,nc,grid,vis,area,rdel,cdel,n,m,cnt);
            }
        }

    }
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int n = grid.size();
        int m = grid[0].size();
        int rdel[] = {-1,0,1,0};
        int cdel[] = {0,1,0,-1};
        int count=1;
        vector<vector<int>>vis(n,vector<int>(m,0));
        int area=0;
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                if(grid[i][j]==1 && vis[i][j]!=1)
                {
                    count=0;
                    dfs(i,j,grid,vis,area,rdel,cdel,n,m,count);
                    area = max(area,count);
                }
            }
        }
        return area;
    }
};
